### PR TITLE
Add around_publish and around_perform

### PR DESCRIPTION
**Pull request by eric-kansas**
add around_publish hook for workers
- wrap `@chore_publisher.publish(self.class.prefixed_queue_name,publish_job_hash)`
- allows for updating the `job_hash` (actual hash payload pushed to queue)

add around_perform hook for jobs
- wrap `perform_job(klass, message)`

## JIRA
https://tapjoy.atlassian.net/browse/DFUN-53

### CODE REVIEW

**Review by Ceraunograph — DISMISSED** [review]

File: lib/chore/worker.rb
Comment (alanbrent) [inline lib/chore/worker.rb:115]: Stray newline here

**Review by alanbrent — DISMISSED** [review]
LGTM pending removal of stray newline

Please don't merge until you do the following:

- [ ] cut a gem from the current code
- [ ] validate it works in production as a gem upgrade with no code that uses the new functionality
- [ ] validate the new functionality works in production

Once that's done, ping me and I'll merge the PR and run through the release process.

**Comment by eric-kansas** [comment]
@alanbrent steps above have been completed. Ready for merge and release process.

**Review by alanbrent — APPROVED** [review]

**Pull request merged**
